nodejs/lib/util/webstorage.js

```diff
@@ -0,0 +1,109 @@
+"use strict";
+var WebStorage = (function() {
+	var path = require('path');
+	var os = require('os');
+	var mkdirs = require('node-mkdirs');
+	var JSONStorage = require('node-localstorage').JSONStorage;
+
+	function storedItem(value, ttl) {
+		var item = {
+			value: value
+		};
+		if(ttl) {
+			item.expires = (Date.now() + ttl);
+		}
+		return item;
+	}
+
+	function valid(item) {
+		var expires = item.expires;
+		return !expires || (Date.now() < expires);
+	}
+
+	function WebStorage(client) {
+		this.client = client;
+		this.storagePath = null;
+		this.localStorage = null;
+		this.sessionStorage = new Map();
```

Comment:
Note that `Map()` is node 4 or later. But looking now, ably-js hasn't worked on node pre-4 for a while, and no-one seems to have noticed. So I think this is fine, we should just update the readme to note that we support nodejs 4 or later, and remove 0.12 from .travis.yml.

Reply:
Yes, I think we can be fairly aggressive in retiring support for older node versions (except where there are dependencies from things like Lambda where end-users have no control).